### PR TITLE
zed-editor: 0.165.4 -> 0.166.1

### PR DESCRIPTION
**Pull request by niklaskorz**
As https://github.com/zed-industries/zed/pull/21550 has finally been merged, we can prepare for one of the next releases making use of the Livekit Rust SDK for macOS and Linux, instead of just the Livekit Swift SDK for macOS.

This PR provides the necessary work to build the Livekit Rust SDK for Zed.

The from-source built of WebRTC itself, which is required by Livekit, has been implemented by @WeetHet, with some feedback and fixes from my side (the commit log reflects this, where I specified @WeetHet as the main author of the WebRTC commit, with me as co-author). Thank you!

Note: Screensharing and audio recording is currently disabled on Linux. This is an upstream limitation that also exists in the official Zed distributions. You can receive audio and screensharings shared by macOS users though.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bbigras** [comment]
v0.166.1 is out.

**Review by jcdickinson — APPROVED** [review]
Smoke tested on NixOS, boots to UI just fine.

**Comment by niklaskorz** [comment]
macOS is pretty much ready, checking Linux again tomorrow

**Comment by jcdickinson** [comment]
Did a smoke test with some light editing, but no collaboration. Seems to be working on NixOS unstable.

**Comment by WeetHet** [comment]
> Did a smoke test with some light editing, but no collaboration. Seems to be working on NixOS unstable.
> 
> Thanks for figuring out LiveKit BTW.

I really would love to also have dynamically linked boringssl in the livekit build cause I really don't love statically linking SSL. Probably not in this PR though

**Comment by niklaskorz** [comment]
> ### `aarch64-linux`
> ❌ 2 packages failed to build:
> 
>     * livekit-libwebrtc
> 
>     * zed-editor

Oh interesting, I'll try to figure out what's going wrong there.

File: pkgs/by-name/li/livekit-libwebrtc/package.nix [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+  meta = with lib; {
+    description = "WebRTC library used by livekit";
+    homepage = "https://github.com/livekit/rust-sdks/";
+    license = licenses.bsd3;
+    maintainers = with lib.maintainers; [
+      WeetHet
+      niklaskorz
+    ];
+    platforms = platforms.linux ++ platforms.darwin;
+  };
```
[the remark this answers is not in the document]
Comment (jansol) [reply pkgs/by-name/li/livekit-libwebrtc/package.nix:297]: isn't the point of the `with lib;` to not need those?

[the remark this answers is not in the document]
Comment (niklaskorz) [reply pkgs/by-name/li/livekit-libwebrtc/package.nix]: Done, thx

**Comment by niklaskorz** [comment]
Overall I think this is ready. There are some more improvements that could be done to the updateScript of livekit-webrtc (currently it works directly on the branch name while it should fetch the latest commit rev of the m114 branch instead and pass that to gclient2nix), and I'm also not entirely sure about the `version = "m114";` (should it be `version = "114-unstable-2024-06-05";` instead? WebRTC does not have tagged versions).

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Pull request merged**